common/src/main/java/feast/common/logging/AuditLogger.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.common.logging;
+
+import com.google.protobuf.Message;
+import feast.common.logging.config.LoggingProperties;
+import feast.common.logging.config.LoggingProperties.AuditLogProperties;
+import feast.common.logging.entry.ActionAuditLogEntry;
+import feast.common.logging.entry.AuditLogEntry;
+import feast.common.logging.entry.AuditLogEntry.FeastInstance;
+import feast.common.logging.entry.AuditLogEntryKind;
+import feast.common.logging.entry.LogResource;
+import feast.common.logging.entry.LogResource.ResourceType;
+import feast.common.logging.entry.MessageAuditLogEntry;
+import feast.common.logging.entry.TransitionAuditLogEntry;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Marker;
+import org.slf4j.MarkerFactory;
+import org.slf4j.event.Level;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class AuditLogger {
+  private static final Marker AUDIT_MARKER = MarkerFactory.getMarker("AUDIT_MARKER");
+  private static FeastInstance instance;
+  private static AuditLogProperties properties;
+
+  @Autowired
+  public AuditLogger(FeastInstance instance, LoggingProperties properties) {
```

Comment:
Can this be simplified by pulling in FeastProperties instead of needing a FeastInstance and LoggingProperties and a class to create a FeastInstance Bean?

Reply:
`FeastProperties` are actually separate classes each coupled to Feast Core or Feast Serving. To do this we would have to introduce a base class/interface that the base classes inherit from.

Reply:
Yea lets just use logging properties, but we don't need to have FeastInstance any more right?

Reply:
FeastInstance will still be constructed in the individual FeastProperties classes